profitbricksapi/profitbricksapi.php

```diff
@@ -0,0 +1,259 @@
+<?php
+/**
+ * @file
+ * @version 0.1
+ * @copyright 2017 CN-Consult GmbH
+ * @author Jens Stahl <jens.stahl@cn-consult.eu>
+ *
+ * License: Please check the LICENSE file for more information.
+ */
+
+namespace ProfitBricksApi;
+
+use Exception;
+
+/**
+ * Class ProfitBricksApi connects to the official ProfitBricksApi and returns easy usable objects.
+ */
+class ProfitBricksApi
+{
+    private $user = "";
+    private $password = "";
+    const profitBricksSnapshotApi = "https://api.profitbricks.com/cloudapi/v3/snapshots?depth=1";
+    const profitBricksDataCenterApi = "https://api.profitbricks.com/cloudapi/v3/datacenters?depth=2";
+    const profitBricksServerApi = "https://api.profitbricks.com/cloudapi/v3/datacenters/[data-center-id]/servers?depth=2";
+    const profitBricksDiskApi = "https://api.profitbricks.com/cloudapi/v3/datacenters/[data-center-id]/servers/[server-id]/volumes?depth=1";
+    const profitBricksDeleteSnapshot = "https://api.profitbricks.com/cloudapi/v3/snapshots/[snapshot-id]";
+    const profitBricksCreateSnapshot = "https://api.profitbricks.com/cloudapi/v3/datacenters/[data-center-id]/volumes/[volume-id]/create-snapshot";
+
+    public function __construct()
+    {
+    }
+
+    /**
+     * Reads all information about snapshots from ProfitBricks into usable objects.
+     *
+     * @return Snapshot[]|bool
+     */
+    public function snapshots()
+    {
+        $snapshots = false;
+        $profitBricksSnapshots = $this->readFromProfitBricks(self::profitBricksSnapshotApi);
```

Comment:
As dicussed (and not a review request), I would prefer an API which returns empty arrays instead of bool|array.
This would reduce this method to this:

```
$snapshots = array();
foreach ($this->readFromProfitBricks(self::profitBricksSnapshotApi) as $snapshot)
{
     $snapshots[$snapshot->id] = new Snapshot($snapshot);
}
return $snapshots;
```